firehose-writer/src/main/java/com/expedia/www/haystack/pipes/firehoseWriter/BlockingQueueEntry.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 Expedia, Inc.
+ *
+ *       Licensed under the Apache License, Version 2.0 (the "License");
+ *       you may not use this file except in compliance with the License.
+ *       You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *       Unless required by applicable law or agreed to in writing, software
+ *       distributed under the License is distributed on an "AS IS" BASIS,
+ *       WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *       See the License for the specific language governing permissions and
+ *       limitations under the License.
+ *
+ */
+package com.expedia.www.haystack.pipes.firehoseWriter;
+
+import com.amazonaws.services.kinesisfirehose.model.PutRecordBatchRequest;
+import com.amazonaws.services.kinesisfirehose.model.PutRecordBatchResult;
+import com.amazonaws.services.kinesisfirehose.model.Record;
+import com.netflix.servo.monitor.Stopwatch;
+
+import java.util.List;
+import java.util.concurrent.Future;
+
+class BlockingQueueEntry {
+    private final Stopwatch stopwatch;
+    private final RetryCalculator retryCalculator;
+    private final PutRecordBatchRequest putRecordBatchRequest;
+    private final Batch batch;
+    private final Future<PutRecordBatchResult> futurePutRecordBatchResult;
+
+    BlockingQueueEntry(Stopwatch stopwatch,
+                       RetryCalculator retryCalculator,
+                       PutRecordBatchRequest putRecordBatchRequest,
+                       Batch batch,
+                       Future<PutRecordBatchResult> futurePutRecordBatchResult) {
+        this.stopwatch = stopwatch;
+        this.retryCalculator = retryCalculator;
+        this.putRecordBatchRequest = putRecordBatchRequest;
+        this.batch = batch;
+        this.futurePutRecordBatchResult = futurePutRecordBatchResult;
+    }
+
+    Stopwatch getStopwatch() {
+        return stopwatch;
```

Comment:
why is getStopWatch defined here? I don't see it being used.

Reply:
It's used at FirehoseConsumer line 80